src/infrastructure/transaction/CreateTransactionFromDTO.ts

```diff
@@ -339,3 +339,42 @@ const extractNetworkType = (version: number): NetworkType => {
 const extractTransactionVersion = (version: number): number => {
     return parseInt(version.toString(16).substr(2, 2), 16);
 };
+
+const extractRecipient = (recipient: string): Address | NamespaceId => {
+        // If bit 0 of byte 0 is not set (like in 0x90), then it is a regular address.
+        // Else (e.g. 0x91) it represents a namespace id which starts at byte 1.
+        const fstByteBit0 = recipient.substr(1, 1);
+
+        if (parseInt(fstByteBit0, 2) === 1) {
+            // namespaceId encoded hexadecimal notation provided
+            // only 8 bytes are relevant to resolve the NamespaceId
+            const relevantPart = recipient.substr(2, 16);
+            return NamespaceId.createFromEncoded(relevantPart);
+        }
+
+        // read address from encoded hexadecimal notation
+        return Address.createFromEncoded(recipient);
+};
+
+const extractMosaics = (mosaics: any): Mosaic[] => {
```

Comment:
Perhaps provide a short description in the document for this private method? as there are some interesting logic here, e.g. the `most significant bit of byte 0 is set`